apps/bot/src/services/streamstats/index.ts

```diff
@@ -0,0 +1,180 @@
+import { prisma, prismaQueue, storage } from "services/db";
+import { TwitchApi } from "services/twitch-api";
+import type { TTwitchApiChatter, TTwitchIrcContext } from "services/types";
+import { logger } from "utils/logger";
+
+export class StreamStatsGatherer {
+  public static instances = new Map<string, StreamStatsGatherer>();
+
+  private api: TwitchApi;
+
+  private activeStream: string | undefined = undefined;
+  private activeStreamId = -1;
```

Comment:
This one is just falllback if something goes really really wrong and doesnt really matter. Whenever stream gathering is started - id is generated and assigned, its actually just helper to avoid calls to db